### PR TITLE
feat: Docs

### LINKED ISSUE #25: Docs

### PR DESCRIPTION
**Pull request by kristianbinau**
## Description

Added Docs

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: src/lib/items/docs.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+			}
+
+			throw new Error(await response.text());
+		}
+
+		return new DocsClass(await response.json());
+	}
+
+	async delete() {
+		const response = await fetch(api('docs/' + this.id), {
```
Comment (frederikpyt) [inline src/lib/items/docs.ts]:
```suggestion
		const response = await fetchFromApi('docs/' + this.id, {
```

File: src/lib/items/docs.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+			}
+
+			throw new Error(await response.text());
+		}
+
+		return new DocsClass(await response.json());
+	}
+
+	async update(name: string, text: string) {
+		const response = await fetch(api('docs'), {
```
Comment (frederikpyt) [inline src/lib/items/docs.ts]: fetchFromApi

File: src/lib/items/docs.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	private _text: string;
+
+	constructor(docsObject: DocsType) {
+		super(docsObject);
+
+		this._text = docsObject.text;
+	}
+
+	static async create(name: string, text: string, parent: FolderType | null) {
+		const response = await fetch(api('docs'), {
```
Comment (frederikpyt) [inline src/lib/items/docs.ts]: fetchFromApi

File: src/pages/u/docs/[id].astro [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import LayoutSidebar from '@layouts/LayoutSidebar.astro';
+// @ts-ignore https://github.com/withastro/language-tools/issues/476
+import { url, fetchFromApi } from '@lib/helpers';
+import { DocsClass } from '@lib/items/docs';
+import DocsEditor from '@components/DocsEditor.vue';
+import { getInitialsByName, getUserColorByInitials } from '@lib/helpers';
+
+const { id } = Astro.params;
+
+// Send a request to the API to get the folder
```
Comment (frederikpyt) [inline src/pages/u/docs/[id].astro]:
```suggestion
// Send a request to the API to get the document
```

**Review by frederikpyt — CHANGES_REQUESTED** [review]
**Found 1 small issue:**
brave_Z68hXPvAK0

Refreshing the test person's page fixes it, and both people seem to show for each other.

File: src/lib/items/docs.ts (hunk 2 above)
Comment (kristianbinau) [reply src/lib/items/docs.ts]:
```suggestion
		const response = await fetchFromApi(api('docs'), {
```

File: src/lib/items/docs.ts (hunk 3 above)
Comment (kristianbinau) [reply src/lib/items/docs.ts]:
```suggestion
		const response = await fetchFromApi(api('docs'), {
```

File: src/lib/items/docs.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ export class DocsClass extends ItemClass {
 	}
 
 	async update(name: string, text: string) {
-		const response = await fetch(api('docs'), {
+		const response = await fetchFromApi(api('docs'), {
```
Comment (frederikpyt) [inline src/lib/items/docs.ts]:
```suggestion
		const response = await fetchFromApi('docs', {
```

File: src/lib/items/docs.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ export class DocsClass extends ItemClass {
 	}
 
 	static async create(name: string, text: string, parent: FolderType | null) {
-		const response = await fetch(api('docs'), {
+		const response = await fetchFromApi(api('docs'), {
```
Comment (frederikpyt) [inline src/lib/items/docs.ts]:
```suggestion
		const response = await fetchFromApi('docs', {
```

**Review by frederikpyt — CHANGES_REQUESTED** [review]

**Review by frederikpyt — APPROVED** [review]
🚀 🌔

**Pull request merged**